pkg/storage/db.go

```diff
@@ -1107,6 +1117,62 @@ func (db *DB) releasePublicIP(ctx context.Context, tx *sql.Tx, arnID string, ip
 	return err
 }
 
+func (db *DB) assignResourceRelationship(ctx context.Context, tx *sql.Tx, arnID string, resource string, when time.Time) error {
+	const assignResourceRelationshipQueryUpdate = `
+update aws_resource_relationship
+set not_before = $1
+where related_arn_id = $2
+  and not_before = to_timestamp(0)
```

Comment:
question: I am slightly confused as to why this query has `to_timestamp(0)`? what is 0 in this context? is this needed?

Reply:
Basically `to_timestamp(0)` is to convert a Unix epoch value to a PostgreSQL timestamp. This implementation is similar to other implementation like `PrivateIpAddress` and `PublicIpAddress`, here we are only over writing not_before if it is set to default `to_timestamp(0)`